### PR TITLE
Fix invalid src url

### LINKED ISSUE #37684: [src] links in beta and nightly documentation are broken
At least some of the `[src]` links in the beta standard library documentation are currently broken. I tried https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html and https://doc.rust-lang.org/beta/std/iter/trait.IntoIterator.html.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #37684.

Thanks to @eddyb's help.

r? @eddyb

### CODE REVIEW

File: src/libcore/macros.rs [hunk 1]
```diff
@@ -550,15 +550,15 @@ macro_rules! unimplemented {
 /// into libsyntax itself.
 ///
 /// For more information, see documentation for `std`'s macros.
-#[cfg(dox)]
-pub mod builtin {
+mod builtin {
```
Comment (eddyb) [inline src/libcore/macros.rs]: This looks wrong, doesn't removing `pub` affect the docs? I could be wrong though.

**Comment by brson** [comment]
Reviewing for backport I don't understand why this patch fixes the linked issue. What does this patch accomplish?

**Comment by brson** [comment]
Please review for backport.

**Comment by eddyb** [comment]
@brson This patch avoids an item (the `mod`) being conditionally present, which changes the compiler-internal DefId's that rustdoc used for linking to source across crates.
It's a minimal hack, reverting the accidental side-effect of adding these docs.
On nightly, #37954 fixed the underlying design problem but is too large to backport.

@rust-lang/tools Any objections to backporting this workaround?

**Comment by alexcrichton** [comment]
Yeah this seems fine to backport.

**Comment by brson** [comment]
Thanks for the explanation @eddyb. Thanks @GuillaumeGomez .

**Pull request merged**